library/Rules/NotEmpty.php

```diff
@@ -17,6 +17,7 @@ public function validate($input)
     {
         if (is_string($input)) {
             $input = trim($input);
+            return isset($input[0]);
```

Comment:
In my opinion this is a side effect of calling a method to validate strings from a integer validation. An integer can't be empty while a string can.